src/Configuration/ConfigurationService.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare(strict_types=1);
+
+namespace DepDoc\Configuration;
+
+use DepDoc\Configuration\Exception\FailedToParseConfigurationFileException;
+use Symfony\Component\Serializer\Encoder\JsonDecode;
+use Symfony\Component\Serializer\Encoder\YamlEncoder;
+use Symfony\Component\Serializer\Normalizer\GetSetMethodNormalizer;
+use Symfony\Component\Serializer\Serializer;
+
+class ConfigurationService
+{
+    /** @var Serializer */
+    protected $serializer;
+    /** @var ConfigurationFileDefinition[] */
+    protected $supportedConfigurationFiles = [];
+
+    /**
+     * @param ConfigurationFileDefinition[] $supportedConfigurationFiles
+     * @param array $normalizers
+     * @param array $encoders
+     */
+    public function __construct(array $supportedConfigurationFiles = [], array $normalizers = [], array $encoders = [])
+    {
+        $this->supportedConfigurationFiles = array_merge([
+            new ConfigurationFileDefinition('.depdoc.json', 'json'),
+            new ConfigurationFileDefinition('.depdoc.yaml', 'yaml'),
+            new ConfigurationFileDefinition('.depdoc.yml', 'yaml'),
+        ], $supportedConfigurationFiles);
+
+        $this->serializer = new Serializer(
```

Comment:
Consider extracting the 2 parameters into single variables before using them. Currently its not 100% clear that the first array is a set of `normalizer` and the second one are real `decoder`.